File: backend/credit/src/main/java/ru/patterns/credit/domain/response/CreditRateModel.java
package ru.patterns.credit.domain.response;

import lombok.AllArgsConstructor;
import lombok.Data;

import java.time.Duration;
import java.util.UUID;

@Data
@AllArgsConstructor
public class CreditRateModel {
    private UUID rateId;
    private String name;
    private int percent;
    private Duration writeOffPeriod;
}

File: mobile/client/app/src/main/java/com/example/g_bankforclient/data/network/model/CreditRateModel.kt
package com.example.g_bankforclient.data.network.model

import java.util.UUID

data class CreditRateModel(
    val rateId: UUID,
    val name: String,
    val percent: Int,
    val writeOffPeriod: String
)

/** Ответ /patterns/api/v1/credit-account/rating/{userId} */
data class CreditRatingNetworkModel(
    val rating: Long,
    val totalCreditCounter: Long,
    val closedCreditCounter: Long,
    val activeCreditAmount: Long,
    val expiredOperationsAmount: Long
)
